dht.go

```diff
@@ -99,7 +99,7 @@ func New(ctx context.Context, h host.Host, options ...opts.Option) (*IpfsDHT, er
 	if err := cfg.Apply(append([]opts.Option{opts.Defaults}, options...)...); err != nil {
 		return nil, err
 	}
-	dht := makeDHT(ctx, h, cfg.Datastore, cfg.Protocols, cfg.BucketSize)
+	dht := makeDHT(ctx, h, cfg.Datastore, cfg.Protocols, cfg.BucketSize, cfg.RoutingTable.LatencyTolerance)
 	dht.autoRefresh = cfg.RoutingTable.AutoRefresh
```

Comment:
While you are on this, would you mind just passing the whole config object to `makeDHT` and reading in the values there ? IMO, what we have right now isn't really neat.

Reply:
Yeah, sure.